avm/res/network/network-interface/main.bicep

```diff
@@ -238,3 +236,161 @@ output resourceGroupName string = resourceGroup().name
 
 @description('The location the resource was deployed into.')
 output location string = networkInterface.location
+
+// ================ //
+// Definitions      //
+// ================ //
+
+@export()
+@description('The resource ID of the deployed resource.')
+type networkInterfaceIPConfigurationType = {
+  @description('Optional. The name of the IP configuration.')
+  name: string?
+
+  @description('Optional. The private IP address allocation method.')
+  privateIPAllocationMethod: ('Dynamic' | 'Static')?
+
+  @description('Optional. The private IP address.')
+  privateIPAddress: string?
+
+  @description('Optional. The resource ID of the public IP address.')
+  publicIPAddressResourceId: string?
+
+  @description('Required. The resource ID of the subnet.')
+  subnetResourceId: string
+
+  @description('Optional. Array of load balancer backend address pools.')
+  loadBalancerBackendAddressPools: backendAddressPoolType[]?
+
+  @description('Optional. A list of references of LoadBalancerInboundNatRules.')
+  loadBalancerInboundNatRules: inboundNatRuleType[]?
+
+  @description('Optional. Application security groups in which the IP configuration is included.')
+  applicationSecurityGroups: applicationSecurityGroupType[]?
+
+  @description('Optional. The reference to Application Gateway Backend Address Pools.')
+  applicationGatewayBackendAddressPools: applicationGatewayBackendAddressPoolsType[]?
+
+  @description('Optional. The reference to gateway load balancer frontend IP.')
+  gatewayLoadBalancer: subResourceType?
+
+  @description('Optional. Whether the specific IP configuration is IPv4 or IPv6.')
+  privateIPAddressVersion: ('IPv4' | 'IPv6')?
+
+  @description('Optional. The reference to Virtual Network Taps.')
+  virtualNetworkTaps: virtualNetworkTapType[]?
+}
+
+@export()
+@description('The type for a backend address pool.')
+type backendAddressPoolType = {
+  @description('Optional. The resource ID of the backend address pool.')
+  id: string?
```

Comment:
Here you can see live what I mean why it will be difficult in the mid-term to uphold stuff like 'resourceId'. Feel free to resolve